pkg/queue/breaker_test.go

```diff
@@ -196,97 +213,139 @@ func TestSemaphore_Acquire_HasNoCapacity(t *testing.T) {
 // Test empty semaphore, add capacity, token can be acquired
 func TestSemaphore_Acquire_HasCapacity(t *testing.T) {
 	gotChan := make(chan struct{}, 1)
+	want := 1
 
-	sem := NewSemaphore(1, 0)
+	sem := newSemaphore(1, 0)
 	tryAcquire(sem, gotChan)
 	sem.Release() // Allows 1 acquire
 
-	assertAcquired(1, gotChan, t)
+	for i := 0; i < want; i++ {
+		select {
+		case <-gotChan:
+			// Successfully acquired a token.
+		case <-time.After(semAcquireTimeout):
+			t.Error("Was not able to acquire token before timeout")
```

Comment:
Does the test make sense after this point, or fatal here? We already didn't get enough tokens, so chances are slim we'll get _more_ tokens that we wanted originally.